We've reached some progress!

```
../src/darwin/REDiskList.h:21:9: fatal error: 'Foundation/Foundation.h' file not found
npm ERR! #import <Foundation/Foundation.h>
npm ERR!         ^~~~~~~~~~~~~~~~~~~~~~~~~
npm ERR! 1 error generated.
```

I pushed another fix attempt. Perhaps more Darwin specific `buildInputs` will be needed now. Naturally it will be much easier with 1st hand access to a Darwin machine. Until then:

@ofborg build balena-cli

--- a/pkgs/tools/admin/balena-cli/default.nix
+++ b/pkgs/tools/admin/balena-cli/default.nix
@@ -7,6 +7,7 @@
 , nodePackages
 , python3
 , udev
+, darwin
 }:
 
 buildNpmPackage rec {
@@ -30,10 +31,15 @@ buildNpmPackage rec {
   nativeBuildInputs = [
     nodePackages.node-gyp
     python3
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.cctools
   ];
 
-  buildInputs = lib.optionals (!stdenv.isDarwin) [
+  buildInputs = lib.optionals stdenv.isLinux [
     udev
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Foundation
+    darwin.apple_sdk.frameworks.Cocoa
   ];
 
   passthru.tests.version = testers.testVersion {

